src/axom/quest/DistributedClosestPoint.hpp

```diff
@@ -0,0 +1,1093 @@
+          };
+
+          auto traversePredicate = [&](const PointType& p,
+                                       const BoxType& bb) -> bool {
+            return squared_distance(p, bb) <= curr_min.minSqDist;
+          };
+
+          // Traverse the tree, searching for the point with minimum distance.
+          it.traverse_tree(qpt, searchMinDist, traversePredicate);
+
+          // If modified, update the fields that changed
+          if(curr_min.minRank == rank)
+          {
+            // SLIC_INFO(
+            //   fmt::format("[On rank {}] Updating dist for particle {} ::"
+            //               "{{ cp: {}->{}; "
+            //               "dist: {}->{}; "
+            //               "rank: {}->{}; "
+            //               "elem: {}->{}}}",
+            //               rank,
+            //               idx,
+            //               query_pos[idx],             //cp
+            //               m_points[curr_min.minElem],
+            //               query_min_dist[idx],        // dist
+            //               sqrt(curr_min.minSqDist),
+            //               query_ranks[idx],           // rank
+            //               curr_min.minRank,
+            //               query_inds[idx],            // index
+            //               curr_min.minElem));
+
+            query_inds[idx] = curr_min.minElem;
+            query_ranks[idx] = curr_min.minRank;
+            query_pos[idx] = m_points[curr_min.minElem];
+
+            //DEBUG
+            query_min_dist[idx] = sqrt(curr_min.minSqDist);
+          }
+          // else
+          // {
+          //   SLIC_INFO(
+          //     fmt::format("[On rank {}] Didn't update dist for particle {} ::"
+          //                 "{{ cp: {}; "
+          //                 "dist: {}; "
+          //                 "rank: {}; "
+          //                 "elem: {}}}",
+          //                 rank,
+          //                 idx,
+          //                 query_pos[idx],
+          //                 query_min_dist[idx],
+          //                 query_ranks[idx],
+          //                 query_inds[idx]));
+          // }
+        }););
+
+    axom::copy(cpIndexes.data(),
+               query_inds.data(),
+               cpIndexes.size() * sizeof(axom::IndexType));
+    axom::copy(cpRanks.data(),
+               query_ranks.data(),
+               cpRanks.size() * sizeof(axom::IndexType));
+    axom::copy(closestPts.data(),
+               query_pos.data(),
+               closestPts.size() * sizeof(PointType));
+
+    // DEBUG
+    axom::copy(minDist.data(),
+               query_min_dist.data(),
+               minDist.size() * sizeof(double));
+  }
+
+private:
+  RuntimePolicy m_runtimePolicy;
+  bool m_isVerbose {false};
+  int m_allocatorID;
+  int m_rank;
+  int m_nranks;
+
+  PointArray m_points;
+  BoxArray m_boxes;
+
+#if defined(AXOM_USE_RAJA) && defined(AXOM_USE_UMPIRE)
+  SeqBVHTree* m_bvh_seq {nullptr};
```

Comment:
I think `spin::BVH` should be usable without RAJA or Umpire in the sequential case.

Reply:
Thanks -- I fixed the guards -- this query no longer requires RAJA or Umpire for the sequential policy